src/ResetMap.jl

```diff
@@ -0,0 +1,287 @@
+import Base: isempty
+
+export ResetMap,
+       get_A,
+       get_b
+
+"""
+    ResetMap{N<:Real} <: LazySet{N}
+
+Type that represents a lazy reset map.
+A reset map is a special case of an affine map ``A x + b, x ∈ X`` where the
+linear map ``A`` is the identity matrix with zero entries in all reset
+dimensions, and the translation vector ``b`` is zero in all other dimensions.
+
+### Fields
+
+- `X`      -- convex set
+- `resets` -- resets (a mapping from an index to a new value)
+
+### Example
+
+```jldoctest resetmap
+julia> X = BallInf([2.0, 2.0, 2.0], 1.0);
+
+julia> r = Dict(1 => 4.0, 3 => 0.0);
+
+julia> rm = ResetMap(X, r);
+
+```
+
+Here `rm` modifies the set `X` such that `x1` is reset to 4 and `x3` is reset to
+0, while `x2` is not modified.
+Hence `rm` is equivalent to the set
+`Hyperrectangle([4.0, 2.0, 0.0], [0.0, 1.0, 0.0])`, i.e., an axis-aligned line
+segment embedded in 3D.
+
+The corresponding affine map ``A x + b`` would be:
+
+```math
+    \begin{pmatrix} 0 & 0 & 0 \\ 0 & 1 & 0 \\ 0 & 0 & 0 \end{pmatrix} x +
+    \begin{pmatrix} 4 & 0 & 0 \end{pmatrix}
+```
+
```

Comment:
"Use the function `get_A` (resp. `get_b`) to create the matrix `A` (resp. vector `b`) corresponding to a given reset map."